crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -759,11 +763,63 @@ impl<
         }
     }
 
+    fn build_num_add_checked(
+        &mut self,
+        dst: &Symbol,
+        src1: &Symbol,
+        src2: &Symbol,
+        num_layout: &Layout<'a>,
+    ) {
+        use Builtin::Int;
+
+        let buf = &mut self.buf;
+
+        let fields = [*num_layout, Layout::bool()];
+        let return_layout = Layout::struct_no_name_order(&fields);
+        let struct_size = return_layout.stack_size(self.target_info);
+
+        let base_offset = self.storage_manager.claim_stack_area(dst, struct_size);
+
+        match num_layout {
+            Layout::Builtin(Int(IntWidth::I64 | IntWidth::I32 | IntWidth::I16 | IntWidth::I8)) => {
+                let dst_reg = self
```

Comment:
Instead of directly using tmp symbols, you can also use `self.storage_manager.with_tmp_general_reg`. Gets rid of the need of manual freeing. Though I guess that would require more nesting.